test/soake2erekt/kafkasource_soak_test.go

```diff
@@ -0,0 +1,334 @@
+	f := feature.NewFeatureNamed("verify-dispatcher-cm-clean")
+	f.Assert("no source from test namespace left in dispatcher cm", func(ctx context.Context, t feature.T) {
+		ns := environment.FromContext(ctx).Namespace()
+		cmlist, err := kubeclient.Get(ctx).CoreV1().ConfigMaps("knative-eventing").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Errorf("error listing knative-eventing configmaps: %v", err)
+			return
+		}
+
+		for _, cm := range cmlist.Items {
+			if strings.HasPrefix(cm.Name, "kafka-source-dispatcher-") {
+
+				dataBytes := cm.BinaryData["data"]
+				if dataBytes != nil {
+					var u map[string]interface{}
+					err = json.Unmarshal(dataBytes, &u)
+					if err != nil {
+						t.Errorf("error unmarshalling %s configmap: %v", cm.Name, err)
+						return
+					}
+
+					uResources, found, err := unstructured.NestedSlice(u, "resources")
+					if !found {
+						// could be still empty?
+						continue
+					}
+					if err != nil {
+						t.Errorf("error getting .resources from %s configmap: %v", cm.Name, err)
+					}
+
+					for _, uResource := range uResources {
+						uResourceMap, ok := uResource.(map[string]interface{})
+						if !ok {
+							t.Errorf("unexpected type of a `resources` item in %s configmap", cm.Name)
+							continue
+						}
+						uNamespace, found, err := unstructured.NestedString(uResourceMap, "reference", "namespace")
+						if !found {
+							// could be not set?
+							continue
+						}
+						if err != nil {
+							t.Errorf("error getting .reference.namespace from %s configmap: %v", cm.Name, err)
+						}
+
+						// t.Logf("XXX reference namespace: %s", uNamespace)
+
+						if uNamespace == ns {
+							t.Errorf("Found reference to a resource in the test namespace %q in the %s configmap", uNamespace, cm.Name)
+						}
+					}
+				}
+			}
+		}
+	})
+
+	return f
+}
+
+/*
+TestKafkaSourceStableSoak on each iteration,
+sends an event to a kafkasource
+verifies an event is received
+*/
+func TestKafkaSourceStableSoak(t *testing.T) {
+	t.Parallel()
+
+	topicPrefix := feature.MakeRandomK8sName("topic")
+	kafkaSinkPrefix := topicPrefix
+	kafkaSourceName := feature.MakeRandomK8sName("kafkaSource")
+	senderPrefix := feature.MakeRandomK8sName("sender")
+
+	namesFn := func(env SoakEnv) kafkaSourceScenarioNames {
+		return kafkaSourceScenarioNames{
+			receiver:    "receiver",
+			sender:      fmt.Sprintf("%s-%d-%d", senderPrefix, env.CopyID(), env.Iteration()),
+			kafkaTopic:  fmt.Sprintf("%s-%d", topicPrefix, env.CopyID()),
+			kafkaSink:   fmt.Sprintf("%s-%d", kafkaSinkPrefix, env.CopyID()),
+			kafkaSource: kafkaSourceName,
+		}
+	}
+
+	soakTest := SoakTest{
+		NamespacePrefix: "test-kafka-source-stable-",
+		SetupFn: func(ctx context.Context, env environment.Environment, t *testing.T) {
+			// TODO: These are actually equivalent to just env.Test(ctx, t, eventshubReceiverFeature(namesFn(SoakEnvFromContext(ctx)))) , so not sure if we should bother...
```

Comment:
I would prefer what you have in the comment. For anyone familiar with the REKT framework, it's immediately clear what's happening. Having RunSoakFeatureFnWithMapping is additional layer of abstraction which doesn't bring much, it's just harder to understand the code.